### DISCUSSION TITLE
How to update a DynamicPartitionsDefinition keys with an existing asset value?

### DISCUSSION THREAD

**Discussion by toddpi314** [dagster-io/dagster#14337] [Q&A]
I want to achieve two things:
1. Have an asset materialization trigger an asset_sensor (no issues, docs sample works fine)
2. On the sensor fire, get the latest value from the asset, and use that to seed new values into the DynamicPartitionsDefinition

I found a series of examples, but many are outdated.

**Comment by jamiedemaria** [comment]
hey @toddpi314  can you share the docs you found and which specific part of step 2 that you're trying to figure out? if you have code snippets of what you've tried and how it hasn't worked that would also be useful

**Reply by toddpi314** [reply]
Starting with https://docs.dagster.io/concepts/partitions-schedules-sensors/partitions#dynamically-partitioned-assets, we are looking to get the materialized asset in the image_sensor implementation.

We have another asset that generates keys that we would like to fetch, and push into the partition def.

**Comment by toddpi314** [comment]
Update:
we figured out how to do this using the `materialize` API from the examples folder.

```
   result = materialize(
        [feature_keys],
        instance=context.instance,
    )
    _feature_keys = result.asset_value("feature_keys")
```
